### DISCUSSION TITLE
--invert-match works only on lines and not with filenames. is that supposed behaviour?

### DISCUSSION THREAD

**Discussion by saifali-dev** [BurntSushi/ripgrep#1795] [Q&A] [answered]
I have 2 files in my dir.
1. Shortcuts.txt (It contains word 'below')
2. Themes & Fonts.txt (it doesn't contains word below)

With simple search **rg below** the output is correct.

When searched like this **rg below --files-with-matches** the output is again correct (as expected only the filename[Shortcuts.txt] is shown).

Now, when searched like this **rg below --files-with-matches --invert-match**. the output contains both filenames. However, it
should return only **Themes & Fonts.txt** because this is the file which doesn't have the word **below** in it.

![image](https://user-images.githubusercontent.com/54573967/107099411-68743880-6833-11eb-8fec-ed81e8b33011.png)
I don't know weather it is bug or expected behaviour, so please ignore the dirname (_rg-bug_).

**Comment by BurntSushi** [comment, marked as answer]
There isn't enough info here to precisely answer. Assuming that `Shortcuts.txt` has at least one line that contains `below`, then the output is correct. `--invert-match` inverts matching in a line oriented way. That is, a matching line is any line that doesn't match the pattern. Thus, any file that contains _any_ line that doesn't match the pattern is considered a matching file by `--files-with-matches`.

This is why the `--files-without-match` flag exists. It is likely what you want here.

**Reply by saifali-dev** [reply]
**Exactly!** this is what I wanted (Now it only prints **Themes & Fonts.txt**)
I have read the whole **User Guide.md** also the full **--help** version of ripgrep. But just skipped the flag you mentioned.
_Thanks a lot!_


**Jazzak Allah**
